SoftwareGuide/Latex/Appendices/CodingStyleGuide.tex

```diff
@@ -1428,6 +1430,29 @@ \section{Pipelines}
 
 \end{itemize}
 
+\section{The auto Keyword}
+\label{sec:TheAutoKeyword}
+
+Available since C++11, the \code{auto} keyword specifies that a variable's type
+is automatically deduced from its initializer.
+
+The \code{auto} keyword should be used to specify a type in the following cases:
+
+\begin{itemize}
+\item The type is duplicated on the left side of an assigment when it is mandated on the
+      right side, e.g. when there is an explicit cast or initializing with \code{new}.
+\item When obtaining container elements, when the element type is obvious from the type
+      of the container.
+\item When declaring iterators in a \code{for} loop.
+\end{itemize}
+
+All other cases should not use \code{auto}, but a semantic type name
+should be used that convey's meaning, as described in Section~\ref{sec:NamingConventions}
+and Section~\ref{subsec:NamingTypedefs}.
+
+Application or example code that uses ITK, as opposed to the toolkit itself,
+may use \code{auto} more liberally for brevity.
```

Comment:
Please change "liberally for brevity" to "liberally."

Line #1454 implies that auto's purpose is "for Brevity".

There are good reasons to use auto that improve performance, readability, and improve communication.  Being explict (long-winded & redundant) can *impede* comprehension of the code.   A good discussion of why using auto can improve performance, re-usability, and improve "code to the interface not the implementaton" policies can be found at "https://herbsutter.com/2013/08/12/gotw-94-solution-aaa-style-almost-always-auto/".

Reply:
**Laziness and commitment**

First, laziness: One common concern is that “writing auto to declare a variable is primarily about saving typing.” However, this is just a misunderstanding of auto. As we saw in GotW #92 and #93 and will see again below, the main reasons to declare variables using auto are for correctness, performance, maintainability, and robustness—and, yes, convenience, but that’s in last place on the list.

Reply:
Done.